src/dune_pkg/rev_store.ml

```diff
@@ -177,11 +177,7 @@ let add_repo ({ lock; dir } as t) ~source =
   let handle = source |> Dune_digest.string |> Dune_digest.to_string in
   Fiber.Mutex.with_lock lock ~f:(fun () ->
     let exists = remote_exists ~dir ~name:handle in
-    let () =
-      match exists with
-      | true -> ()
-      | false -> remote_add ~dir handle source
-    in
+    if not (remote_exists ~dir ~name:handle) then remote_add ~dir handle source;
```

Comment:
The suggestion I made before didn't let me edit the line above. You need to also remove that.